tests/integration/Owncloud/OcisPhpSdk/DriveTest.php

```diff
@@ -72,15 +73,16 @@ public function testDeleteEnabledDrive(): void
 
     public function testGetDriveRole(): void
     {
-        // At the time of writing, "stable" is major version 5 of ocis.
-        // This functionality works with major version 6.
-        // When ocis major version 6 has been released as "stable" then remove this test skip.
-        if (getenv('OCIS_VERSION') === "stable") {
+        try {
+            $role = $this->drive->getRoles();
+        } catch(EndPointNotImplementedException) {
+            if (getenv('OCIS_VERSION') !== "stable") {
+                $this->fail("EndPointNotImplementedException was thrown unexpectedly");
+            }
             $this->markTestSkipped(
                 'This test is skipped because root endpoint for drive share is not applicable for version 5 of OCIS.'
             );
```

Comment:
We do not want to skip the test. At this point in the code we are running on "stable" and we got `EndPointNotImplementedException`. That is what the system is supposed to do. We can let the test finish "normally".

Reply:
Or you can refactor the code to do the 2 different test flows more separately, or make 2 separate test cases.

Whatever is done, we want tests that pass in each environment.